gui/app/simple_clock.cpp

```diff
@@ -0,0 +1,61 @@
+#include "simple_clock.h"
+#include <QTime>
+
+SimpleClock::SimpleClock(QObject *parent)
+    : QObject(parent), enabled_(true), update_interval_(1000), hour_(0),
+      minute_(0), second_(0), timer_(this) {
+
+  connect(&timer_, &QTimer::timeout, this, &SimpleClock::update_timer);
+  timer_.start();
+}
+
+SimpleClock::~SimpleClock() { timer_.stop(); }
+
+bool SimpleClock::GetEnabled() const { return enabled_; }
+
+int SimpleClock::GetUpdateInterval() const { return update_interval_; }
+
+int SimpleClock::GetHour() const { return hour_; }
+
+int SimpleClock::GetMinute() const { return minute_; }
+
+int SimpleClock::GetSecond() const { return second_; }
+
+void SimpleClock::SetUpdateInterval(int update_interval) {
+  if (update_interval_ != update_interval) {
+    update_interval_ = update_interval;
+    emit updateIntervalChanged();
```

Comment:
Do we need the ability to change the update interval and enabled? Seems like if we keep them fixed, then the code becomes much simpler.

Reply:
Probably not necessary, but also something trivial. I can remove it if you prefer.